logdevice/server/read_path/ServerReadStream.h

```diff
@@ -255,6 +257,16 @@ class ServerReadStream : boost::noncopyable {
     return flow_group.level(getReadPriority());
   }
 
+  void updateTrafficClass(
+      std::chrono::milliseconds traffic_class_updation_threshhold) {
+    if (last_delivered_timestamp_ + traffic_class_updation_threshhold >
+        SystemTimestamp::now().toMilliseconds()) {
+      traffic_class_ = TrafficClass::READ_TAIL;
+    } else {
+      traffic_class_ = TrafficClass::READ_BACKLOG;
+    }
+  }
+
```

Comment:
changes should only be applied if traffic_class_ was not previously set by client.
We don't want automation to override client set values

Reply:
Can you please help me with how to do this?
should I add a variable which remembers if "setTrafficClass()" has  been called?